### ISSUE TITLE
very strange behavior - factory not registered based on name of file

### ISSUE THREAD

**Issue by soynog** [thoughtbot/factory_girl#993]
I keep getting the following error:

```
Factory not registered: zipcode (ArgumentError)
```

I have a model called Zipcode in a rails app using `factory_girl_rails`. When I create a factory file at `factories/zipcodes.rb`, I get the error.
I've tried stopping spring, preparing the db, adding `FactoryGirl.find_definitions` to my config file, etc. Nothing has worked; and other nearly identical models (City, County) with different names work fine.

Further perplexing: If I define the factory in a different file, `factories/geographies.rb`, specifying its class in the factory definition, it works great. BUT (and here's the crazy part) if I call that file `zgeographies.rb`, it breaks! Same thing with `ygeographies.rb`,  `xgeographies.rb` ... all the way through `rgeographies.rb`. But `ageographies.rb` through `qgeographies.rb` work fine! I've got other factories that start with various letters of the alphabet all the way through `w`. They all work fine.

Not urgent because I've got it working, but I'm just baffled by this and would love an answer. Thanks!

**Comment by sngeth** [comment]
I can't seem to reproduce your issue with a new app. Any other app details?

**Comment by soynog** [comment]
Hmm. I just tried recreating the issue myself with a zipcodes.rb and a zgeographies.rb, and it's working fine now for me too. So I don't know what's going on.

Only things I can think of that might be relevant are that I created the various files manually (rather than using generators, aside from the migrations), along with a couple other new tables. Also they're tables with geospatial fields. Any particular details that might be helpful?

Maybe it has something to do with spring not loading the new files properly? I did try stopping and starting it a few times... I just don't understand why it would have worked fine for some filenames and not for others.

Anyway, it's a moot point for me now, so feel free to close this issue.
